### PR TITLE
Disallow trusting self

### LINKED ISSUE #1020: Trusting yourself should fail
This might need a feature flag.

### PR DESCRIPTION
**Pull request by vogel**
Fixes #1020
Docs available: https://github.com/stellar/docs/compare/master...vogel:disallow-trusting-self (not sure about policy of merging docs).

### CODE REVIEW

File: src/transactions/AllowTrustOpFrame.cpp [hunk 1]
```diff
@@ -29,6 +29,17 @@ bool
 AllowTrustOpFrame::doApply(Application& app, LedgerDelta& delta,
                            LedgerManager& ledgerManager)
 {
+    if (app.getLedgerManager().getCurrentLedgerVersion() > 2)
```
Comment (jedmccaleb) [inline src/transactions/AllowTrustOpFrame.cpp]: Shouldn't you use the ledgerManager that is passed in?

Comment (vogel) [inline src/transactions/AllowTrustOpFrame.cpp]: Updated

**Comment by vogel** [comment]
Changed target branch

File: src/transactions/AllowTrustTests.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
+// Copyright 2014 Stellar Development Foundation and contributors. Licensed
+// under the ISC License. See the COPYING file at the top-level directory of
+// this distribution or at http://opensource.org/licenses/ISC
```
Comment (raymens) [inline src/transactions/AllowTrustTests.cpp]: Is there a specific reason this file is licensed under ISC instead of APACHE?

Comment (vogel) [inline src/transactions/AllowTrustTests.cpp]: thanks for noticing, I've copied the licence for this file from ManageDataTests.cpp
I'll change that and ask Jed about the ManageDataTests.cpp file

**Pull request merged**